docs/user/guides/build/skip-build.rst

```diff
@@ -0,0 +1,263 @@
+
+When to skip builds
+-------------------
+
+There are several scenarios where you might want to skip documentation builds:
+
+**Save resources on irrelevant changes**
+   Skip builds when changes don't affect documentation,
+   such as changes only to source code, tests, or CI configuration files.
+
+**Avoid redundant builds**
+   Skip builds for draft pull requests, work-in-progress branches,
+   or commits with specific markers like ``[skip ci]`` in the commit message.
+
+**Conditional documentation updates**
+   Only build documentation when specific files or directories are modified,
+   such as the ``docs/`` folder or configuration files.
+
+**Branch-specific logic**
+   Skip builds on certain branches that don't require documentation updates,
+   such as experimental or development branches.
+
+Examples
+--------
+
+The following examples demonstrate common use cases for skipping builds.
+All examples use the :ref:`config-file/v2:build.jobs` configuration key
+in your ``.readthedocs.yaml`` file.
+
+Skip builds when documentation files haven't changed
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This example skips pull request builds when there are no changes to documentation-related files
+compared to the ``main`` branch:
+
+.. code-block:: yaml
+   :caption: .readthedocs.yaml
+
+   version: 2
+   build:
+     os: "ubuntu-22.04"
+     tools:
+       python: "3.12"
+     jobs:
+       post_checkout:
+         # Cancel building pull requests when there aren't changes in the docs directory or YAML file.
+         # You can add any other files or directories that you'd like here as well,
+         # like your docs requirements file, or other files that will change your docs build.
+         #
+         # If there are no changes (git diff exits with 0) we force the command to return with 183.
+         # This is a special exit code on Read the Docs that will cancel the build immediately.
+         - |
+           if [ "$READTHEDOCS_VERSION_TYPE" = "external" ] && git diff --quiet origin/main -- docs/ .readthedocs.yaml;
+           then
+             exit 183;
+           fi
+
+You can customize this example by:
+
+* Adding more paths to check: ``docs/ .readthedocs.yaml requirements/docs.txt``
+* Checking against a different branch: ``origin/develop`` instead of ``origin/main``
+* Using different comparison operators to check for specific file patterns
+
+Skip builds based on commit message
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This example skips builds when the commit message contains ``[skip ci]`` or ``[ci skip]``:
+
+.. code-block:: yaml
+   :caption: .readthedocs.yaml
+
+   version: 2
+   build:
+     os: "ubuntu-22.04"
+     tools:
+       python: "3.12"
+     jobs:
+       post_checkout:
+         # Check if the latest commit message contains "[skip ci]" or "[ci skip]"
+         - (git --no-pager log --pretty="tformat:%s -- %b" -1 | grep -viq "\[skip ci\]\|\[ci skip\]") || exit 183
```

Comment:
This command was changed for some reason.


```suggestion
         - (git --no-pager log --pretty="tformat:%s -- %b" -1 | paste -s -d " " | grep -viq "skip ci") || exit 183
```

I don't want to change it since we know the previous one works fine.